src/Compilers/Core/Portable/InternalUtilities/FourState.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Diagnostics;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Represents an optional bool? as a single byte.
+    /// </summary>
+    internal enum FourState : byte
```

Comment:
>FourState [](start = 18, length = 9)

Consider moving this to a `private` type in `SourceModuleSymbol` if that's the only place this is needed.